src/hammer-vlsi/technology/asap7/__init__.py

```diff
@@ -191,4 +193,40 @@ def scale_gds_script(self, gds_file: str) -> str:
 gds_lib.write_gds('{gds_file}')
         """.format(cell_list_file=os.path.join(self.extracted_tarballs_dir, 'ASAP7_PDKandLIB.tar/ASAP7_PDKandLIB_v1p5/asap7libs_24.tar.bz2/asap7libs_24/gds/cell_list.txt'), gds_file=gds_file)
 
+    def get_tech_par_hooks(self, tool_name: str) -> List[HammerToolHookAction]:
+        hooks = {"innovus": [
+            HammerTool.make_persistent_hook(asap7_innovus_settings),
+            HammerTool.make_post_insertion_hook("write_design", scale_final_gds)
+            ]}
+        return hooks.get(tool_name, [])
+
+def asap7_innovus_settings(ht: HammerTool) -> bool:
+    assert isinstance(ht, HammerPlaceAndRouteTool), "Innovus settings only for par"
+    assert isinstance(ht, CadenceTool), "Innovus is Cadence"
+    """Settings that need to be reapplied at every tool invocation"""
+    ht.append('''
+set_db route_design_bottom_routing_layer 2
```

Comment:
Are these settings not persistent?

Also, are these settings only "dropped" for this technology or is it the same across all technologies?

Reply:
To be honest, I am not certain what does and does not persist, and there's no Innovus documentation indicating what gets dropped and why. I also have no idea if it varies by technology without much trial-and-error. That said, it doesn't hurt to re-apply these settings; there are reference flows that will always apply general Innovus database settings at every tool invocation, so I opted to just be cautious here.

Reply:
What does a `get_db` return if resumed? I agree it's good to do it well, just want to make sure if we put in code we know why it's the case.

Reply:
For these two particular settings, `get_db` does have them already, so persistence is not strictly necessary in this case. However, I think we can leave this for now, as it serves as an example for people building tech plugins.

Reply:
I see. My thought is that if we don't need it it's better to try and keep the plugin lean...

Reply:
From offline discussion with @edwardcwang, this will remain persistent but it is documented that persistence is not truly necessary for these particular commands; it instead serves as an example for other tech plugins for how to implement persistence.